test/daplink_firmware.py

```diff
@@ -80,7 +80,19 @@ class ProjectFirmwareBundle(firmware.FirmwareBundle):
     def __init__(self, directory, tool):
         tool_dir = os.path.abspath(directory + os.sep + 'projectfiles' +
                                    os.sep + tool)
+        errmsg = "Error: DAPLink projects are missing. Did " \
+                   "you forget to generate them by running " \
+                   "'progen generate -t %s'?\nExpecting them at %s" % (tool, tool_dir)
+
+        if not os.path.exists(tool_dir): 
+            print (errmsg)
+            exit(-1) 
+
         project_dir_list = os.listdir(tool_dir)
+        if not project_dir_list:
+            print (errmsg)
+            exit(-1) 
```

Comment:
Sounds good. I will leave it as is. Doesn't seem like a recoverable error condition. Like you said, we can change that as needed

John

> On Apr 14, 2016, at 3:34 PM, Russ Butler <notifications@github.com> wrote:
> 
> In test/daplink_firmware.py:
> 
> >          project_dir_list = os.listdir(tool_dir)
> > +        if not project_dir_list:
> > +            print (errmsg)
> > +            exit(-1) 
> By script I meant run_test.py. Currently everything which uses exit() resides in run_test.py and isn't meant to be used outside that script. This is more of a nitpick so don't feel like it needs to change. If the need arises in the future to make this case recoverable then it can be changed later.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>